shared/middleware/channels_api.rb

```diff
@@ -164,7 +164,9 @@ class ChannelsApi < Sinatra::Base
   # Marks the specified channel as published.
   #
   post %r{/v3/channels/([^/]+)/publish/([^/]+)} do |channel_id, project_type|
-    not_authorized if under_13?
+    not_authorized unless owns_channel?(channel_id)
+    not_authorized unless %w(artist playlab applab gamelab).include?(project_type)
+    not_authorized if under_13? && !%w(artist playlab).include?(project_type)
```

Comment:
> In the owns_channel? and under_13? cases the better response would be forbidden (User is authenticated but not allowed to access resource) rather than not_authorized (User is not authenticated). It would be reasonable to add a fourth case (listed first) that returned not_authorized if the user is not signed in. 

Done for under 13.

I agree in principle with returning a different code depending on whether the user is signed in as a non-owner vs. not signed in, however we'd have to go to the dashboard DB (see [current_user](https://github.com/code-dot-org/code-dot-org/blob/staging/shared/middleware/helpers/auth_helpers.rb#L17-L21)) to determine this, which doesn't seem worth it. That leaves us with the `owns_channel?` check having to return either `forbidden` or `not_authorized`, neither of which will be correct all the time. I'm going to go with `not_authorized` since that's what's used everywhere else for that check, but if we want to change it to `forbidden`, I'd be happy to do that in all the places its used in a follow-up PR.

> I think failing the project_type whitelist should probably return bad_request.

Done.